src/useCameraShake.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import * as THREE from 'three'
+import { Camera, useThree, useFrame } from 'react-three-fiber'
+import { makeNoise2D } from 'open-simplex-noise'
+
+interface ShakeConfig {
+  decay: boolean
+  decayRate: number
+  maxYaw: number
+  maxPitch: number
+  maxRoll: number
+  yawFrequency: number
+  pitchFrequency: number
+  rollFrequency: number
+  yawNoiseSeed: number
+  pitchNoiseSeed: number
+  rollNoiseSeed: number
+}
+export type ShakeConfigPartial = Partial<ShakeConfig>
+
+const defaultConfig: ShakeConfig = {
+  decay: false,
+  decayRate: 0.65,
+  maxYaw: 0.1,
+  maxPitch: 0.1,
+  maxRoll: 0.1,
+  yawFrequency: 1,
+  pitchFrequency: 1,
+  rollFrequency: 1,
+  yawNoiseSeed: 10,
+  pitchNoiseSeed: 20,
+  rollNoiseSeed: 30,
+}
+
+export interface ShakeController {
+  setTrauma: (val: number) => void
+  getTrauma: () => number
+  addTrauma: (val: number) => void
+  clearTrauma: () => void
+}
+
+export function useCameraShake(
+  controlledCam: React.MutableRefObject<Camera | undefined>,
+  shakeConfig?: ShakeConfigPartial,
+  initialTrauma?: number
+) {
+  const { setDefaultCamera } = useThree()
+  const shakyCam = React.useMemo(() => new THREE.PerspectiveCamera(), [])
```

Comment:
If you're changing the default camera to perspective, does that mean that this hook can be only used with `PerspectiveCamera`? Because this is why theres a TS error on L86. The copy source has to be a `PerspectiveCamera` so the hook would probably break with `OrthographicCamera`.

Reply:
Ideally here you should use the default camera from `useThree`.

Reply:
I had a look at the code & did some suggestions @gsimone you can see them here – https://github.com/Code47X/drei/pull/1